### PR TITLE
Readme update

### PR DESCRIPTION
**Pull request by MikeTheCanuck**
Two concerns addressed here:
1. Finally outlining our strategy for enabling deployment of side-by-side staging and production environments.
2. Underprivileged consumers of this code might be stymied by lack of permissions, so I've started to outline the kinds of permissions that are required to successfully deploy this stack to CloudFormation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,7 +107,7 @@ This is specified in the [master.yaml](master.yaml) template.
 
 By default, [t2.large](https://aws.amazon.com/ec2/instance-types/) instances are used, but you can change this by modifying the following section:
 
-```
+``` AWS
```
Comment (DingoEatingFuzz) [inline README.md]: If you use ```yaml here you will get syntax highlighting.

```yaml
VPC:
  Type: AWS::CloudFormation::Stack
    Properties:
      TemplateURL: !Sub ${TemplateLocation}/infrastructure/vpc.yaml
      Parameters:
        EnvironmentName:    !Ref AWS::StackName
        VpcCIDR:            10.180.0.0/16
        PublicSubnet1CIDR:  10.180.8.0/21
        PublicSubnet2CIDR:  10.180.16.0/21
        PrivateSubnet1CIDR: 10.180.24.0/21
        PrivateSubnet2CIDR: 10.180.32.0/21
```

File: README.md [hunk 2]
```diff
@@ -55,9 +61,18 @@ After the CloudFormation templates have been deployed, the [stack outputs](http:
...
+* go to AWS CloudFormation - if creating new stack (e.g. for testing), choose "create stack"; if updating an existing stack, select that stack then click the *Update* button
+
+#### Security requirements
+
+The account of the AWS user who initially creates the stack requires many privileges in AWS, including:
+
+* IAM Role creation
+* IAM Policy creation
+
+Subsequent incremental Updates to an existing stack can sometimes be performed by AWS users with less privileges, depending on which stack objects are being created, updated or deleted.
```
Comment (DingoEatingFuzz) [inline README.md]: Probably worth mentioning that Cloud Formation will automatically rollback a stack change if the user who performed the updated didn't have adequate permissions. Just so people don't think they could possibly derail the state of the world.

File: README.md [hunk 3]
```diff
@@ -13,17 +13,23 @@ A Set of YAML templates for deploying the HackOregon infrastructure on [Amazon E
...
+### Infrastructure-as-Code
 
-This template can be used repeatedly to create identical copies of the same stack (or to use as a foundation to start a new stack).
+This set of templates can be used to create near-identical copies of the same stack (or to use as a foundation to start a new stack).
+
+Master templates correspond to the following deployed clusters in Hack Oregon:
+
+1. `master.yaml` - the historical "hacko-integration" cluster that has been used as test/staging/production since Hack Oregon's 2017 project season.
+2. (Coming soon) `master-staging.yaml` - a dedicated staging environment for all 2017+ Hack Oregon projects.  Looser access to developers, deploys from `develop` branch or equivalent in each project, limited resources to keep costs down.
+3. (Coming soon) `master-production.yaml` - a dedicated production environment for all 2017+ Hack Oregon projects.  Restricted access to developers, only deploys from `master` branch in each project, production-grade resource allocation (greater number of load-balanced tasks, higher Cpu and Memory resource allocation).
```
Comment (DingoEatingFuzz) [inline README.md]: I like it 👍

File: README.md (hunk 1 above)
Comment (MikeTheCanuck) [reply README.md]: duh, like AWS is a language (that anyone *respects* anyway...) thanks for the tip

**Pull request merged**